Orm/Xtensive.Orm/Orm/Building/Builders/PartialIndexFilterBuilder.cs

```diff
@@ -54,13 +83,19 @@ protected override Expression VisitBinary(BinaryExpression b)
       var left = Visit(b.Left);
       var right = Visit(b.Right);
 
-      FieldInfo field;
-      if (entityAccessMap.TryGetValue(left, out field) && IsNull(right))
+      if (entityAccessMap.TryGetValue(left, out var field) && IsNull(right))
         return BuildEntityCheck(field, b.NodeType);
       if (entityAccessMap.TryGetValue(right, out field) && IsNull(left))
         return BuildEntityCheck(field, b.NodeType);
 
       return base.VisitBinary(b);
+
+      static bool EnumRewritableOperations(BinaryExpression b)
+      {
+        return b.NodeType.In(ExpressionType.Equal, ExpressionType.NotEqual,
```

Comment:
That call must be pretty expensive.
It better be a regular boolean expression to avoid allocations.
The implementation below must be way faster.
```csharp
  var nodeType = b.NodeType;
  return nodeType==ExpressionType.Equal || nodeType==ExpressionType.NotEqual || ...;
```

Reply:
Ok